src/main/java/net/minecraftforge/fluids/FluidType.java

```diff
@@ -823,12 +822,7 @@ public final boolean isLighterThanAir()
      */
     public boolean isVaporizedOnPlacement(Level level, BlockPos pos, FluidStack stack)
     {
-        if (level.dimensionType().ultraWarm())
-        {
-            BlockState state = this.getBlockForFluidState(level, pos, this.getStateForPlacement(level, pos, stack));
-            return state != null && state.getMaterial() == Material.WATER;
-        }
-        return false;
+        return level.dimensionType().ultraWarm() && stack.getFluid().getFluidType() == ForgeMod.WATER_TYPE.get();
```

Comment:
For right now, revert this change. This is a logic break which will probably screw up mods which already use the material water to implement it. If we decide to add to this or modify it later, it'll probably be on a future major version where such a logic change is more acceptable along with other changes to the Fluid API.

Reply:
Okay, I will revert this change :)

Reply:
reverted